app/controllers/api/v1/service_plans_controller.rb

```diff
@@ -19,20 +19,20 @@ def create
       end
 
       def show
-        plan = ServicePlan.find(params.require(:id))
-        render :json => plan
+        svc = Catalog::ServicePlanJson.new(:service_plan_id => params.require(:id))
+        render :json => svc.process.json
       end
 
       def base
-        plan = ServicePlan.find(params.require(:service_plan_id))
-        render :json => plan.base
+        svc = Catalog::ServicePlanJson.new(:service_plan_id => params.require(:service_plan_id), :schema => "base")
+        render :json => svc.process.json
       end
 
       def modified
         plan = ServicePlan.find(params.require(:service_plan_id))
 
         if plan.modified.present?
-          render :json => plan.modified
+          render :json => Catalog::ServicePlanJson.new(:service_plan_id => plan.id).process.json
```

Comment:
This kinda sucks, we're doing `ServicePlan.find` on line 32, and then we're doing it again by passing in the ID here and doing it in `Catalog::ServicePlanJson`. Think there's an easy solution for this?

Reply:
Yeah by default it looks up modified, updated to just call the service class and see if the `create_json_schema` field is present. If it's there that means there was a modified, otherwise we 204.